### PR TITLE
Install.sh script

### PR DESCRIPTION
**Pull request by Tylerjet**

### CODE REVIEW

**Comment by Tylerjet** [comment]
As a draft for now, can be tested with ```git pull origin pull/73/head```

**Comment by Tylerjet** [comment]
There are some lines i will like update/move to functions for ease of modifying/editing in the future but it is in a state now where the main actions can be tested.

**Comment by Tylerjet** [comment]
Hehe yeah it looks like a lot but probably 25% of it is just the text that gets pushed into service files lol.

Re: moonraker that should be simple enough, On the second run it would have looked for the moonraker.conf and checked if it already had the klipper backup entry added and if it does it skips that step so that is expected.

I'm not sure about a single step back but could likely do something like a key combination that restarts that configuration section, will see on that one.

**Review by Staubgeborener — APPROVED** [review]

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+[update_manager klipper-backup]
+type: git_repo
+path: ~/klipper-backup
+origin: https://github.com/Staubgeborener/klipper-backup.git
+managed_services: moonraker
+primary_branch: main
+MOONRAKER_CONF
+            sleep .5
+            echo -e "${Y}●${NC} Restarting moonraker."
+            sudo systemctl restart moonraker.service
```
[the remark this answers is not in the document]
Comment (Tylerjet) [reply install.sh]: Not anymore further up that code does not run at all if the moonraker folder doesn't exist

[the remark this answers is not in the document]
Comment (Tylerjet) [reply install.sh]: Line 197

[the remark this answers is not in the document]
Comment (Tylerjet) [reply install.sh]: Though I may make it both the folder exists and the service is found and active

**Comment by Staubgeborener** [comment]
Someone in Minimal3Ds video mentioned, that the timestamp format is inconsistent. Maybe we could also fix this with this PR, as we also create a cronjob with this install script which handles the timestamp format.

image

**Comment by Tylerjet** [comment]
hrmm which macro are they referring to i wonder as i though i had updated them all, but yeah I'll go through and double check they all match the new format of $(date +"%x - %X")

**Comment by Tylerjet** [comment]
ahh yeah i see backup on boot still had %%D instead of the new formatt will be in next commit

**Comment by Tylerjet** [comment]
I really hate the look of whiptail compared to the rest of the script but hey it seems to work, might look at another way to trigger going back that i could go back to all text based.

**Comment by Tylerjet** [comment]
More changes will be coming as i think i found the direction and information i wanted to streamline the process. 🎉

**Comment by Tylerjet** [comment]
@Staubgeborener definitely test from that latest commit

**Comment by Staubgeborener** [comment]
Looks pretty good so far. Maybe add a check for `install_backup_service()`, `install_cron()` and `install_filewatch_service()` if these service are already installed? So only ask to install those services if they aren't already installed.

**Comment by Tylerjet** [comment]
Merging this into the newly created ```installer-dev``` branch instead of into main for now while its worked on but this way its not still sitting as a pending PR.

**Pull request merged**